### PR TITLE
replace brittle regex for ancestor titles; see #45654

### PR DESCRIPTION
**Pull request by mp-jwolkerstorfer**

### CODE REVIEW

File: src/parent_child_title_filter.coffee [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-    fullTitleParts = pattern.exec(obj.full_title)
-    return obj.full_title if !fullTitleParts?
-    parentTitle = "<span class=\"parent-title\">#{fullTitleParts[1]}</span><br>" || null
-    title = "<span class=\"title\">#{fullTitleParts[2]} #{fullTitleParts[3]}</span>"
+    return obj.full_title if obj.full_title == obj.display_title
+    ancestorTitles = obj.full_title.replace(" - #{obj.display_title}", '')
+    parentTitle = null
+    if ancestorTitles
+      parentTitle = "<span class=\"parent-title\">#{ancestorTitles}</span><br>"
+    title = "<span class=\"title\">#{obj.display_title}</span>"
```
Comment (mp-jgoetzinger) [inline src/parent_child_title_filter.coffee]: you can use the `"""` notation here (coffeescript feature)..

```
title = """<span class="title">#{obj.display_title}</span>"""
```

File: src/parent_child_title_filter.coffee [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
     fullTitle = ''
     fullTitle = parentTitle if parentTitle?
```
Comment (mp-jgoetzinger) [inline src/parent_child_title_filter.coffee]:
```
fullTitle = parentTitle || ''
``` 

?

Comment (mp-jgoetzinger) [reply src/parent_child_title_filter.coffee]: or even

```
fulltitle = "#{parentTitle}#{title}"
```

File: src/parent_child_title_filter.coffee (hunk 1 above)
Comment (mp-jwolkerstorfer) [reply src/parent_child_title_filter.coffee]: or i could just switch the double quotes on the class to single quotes?

File: src/parent_child_title_filter.coffee [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
   (obj) ->
-    pattern = /^(.*) - (.*) (\([0-9/].+\))/
-    fullTitleParts = pattern.exec(obj.full_title)
-    return obj.full_title if !fullTitleParts?
-    parentTitle = "<span class=\"parent-title\">#{fullTitleParts[1]}</span><br>" || null
-    title = "<span class=\"title\">#{fullTitleParts[2]} #{fullTitleParts[3]}</span>"
+    return obj.full_title if obj.full_title == obj.display_title
+    ancestorTitles = obj.full_title.replace(" - #{obj.display_title}", '')
+    parentTitle = null
+    if ancestorTitles
```
Comment (mp-jgoetzinger) [inline src/parent_child_title_filter.coffee]: i think this is always `true`

**Comment by mp-jgoetzinger** [comment]
👍

**Pull request merged**